crates/anvil/src/config.rs

```diff
@@ -1032,6 +858,190 @@ latest block number: {latest_block}"
 
         backend
     }
+
+    /// Configures everything related to forking based on the passed `eth_rpc_url`:
+    ///  - returning a tuple of a [ForkedDatabase](ForkedDatabase) and [ClientFork](ClientFork)
+    ///    which can be used in a [Backend](mem::Backend) to fork from.
+    ///  - modifying some parameters of the passed `env`
+    ///  - mutating some members of `self`
+    pub async fn setup_fork_db(
+        &mut self,
+        eth_rpc_url: String,
+        env: &mut revm::primitives::Env,
+        fees: &FeeManager,
+    ) -> (Arc<tokio::sync::RwLock<dyn Db>>, Option<ClientFork>) {
```

Comment:
this should return proper types not Arc<Rwlock>

Reply:
If I return a `ForkedDatabase` or `tokio::sync::RwLock<dyn Db>` from `setup_fork_db` then I'm not sure I can have the `ClientFork` reference the same `db` because that wants a `Arc<AsyncRwLock<ForkedDatabase>>` and I'd have to clone the `db` which AFAIK would make a new instance.

Let me know if I'm missing something.

Reply:
oh right, you're correct of course